Libraries/LibCrypto/Cipher/Mode/CTR.h

```diff
@@ -95,10 +95,14 @@ class CTR : public Mode<T> {
 
     virtual ~CTR() {}
 
+    // Must intercept `Intent`, because AES must always be set to
+    // Encryption, even when decrypting AES-CTR.
+    // TODO: How to deal with other ciphers?
     template<typename... Args>
-    explicit constexpr CTR<T>(Args... args)
-        : Mode<T>(args...)
+    explicit constexpr CTR<T>(const ByteBuffer& user_key, size_t key_bits, Intent intent = Intent::Encryption, Args... args)
+        : Mode<T>(user_key, key_bits, args...)
     {
+        (void)intent;
```

Comment:
You can just say
```c++
explicit constexpr CTR<T>(const ByteBuffer& user_key, size_t key_bits, Intent = Intent::Encryption, Args... args)
```
and get rid of the `(void)intent;`.

Reply:
Lovely! I wasn't aware that C++ can do that.